compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -1939,58 +1392,12 @@ TR::Register *OMR::X86::TreeEvaluator::dRegStoreEvaluator(TR::Node *node, TR::Co
    TR::Register *childGlobalReg = cg->machine()->getFPStackRegister(fpStackSlot);
    TR::Register *globalReg = cg->evaluate(child);
 
-   if (cg->useSSEForDoublePrecision())
-      {
-      if (globalReg->getKind() != TR_FPR)
-         globalReg = TR::TreeEvaluator::coerceFPRToXMMR(child, globalReg, cg);
-
-      machine->setXMMGlobalRegister(globalRegNum - machine->getNumGlobalGPRs(), globalReg);
-      cg->decReferenceCount(child);
-      return globalReg;
-      }
-   else
-      {
-      machine->setFPStackRegister(globalRegNum - machine->getNumGlobalGPRs(), toX86FPStackRegister(globalReg));
+   if (globalReg->getKind() != TR_FPR)
+      globalReg = TR::TreeEvaluator::coerceFPRToXMMR(child, globalReg, cg);
```

Comment:
Delete these two lines. The globalReg can't be anything other than a TR_FPR. You could add a TR_FATAL_ASSERT for this instead.